### PR TITLE
[MM-20431] Migrate components/reset_status_modal module and associated tests to TypeScript

### PR DESCRIPTION
**Pull request by kcc343**
#### Summary
Migrate components/reset_status_modal module and associated tests to TypeScript

I'm kind of new to TypeScript so I was not sure about the proper way to convert the code. Let me know if I need to change anything!

#### Ticket Link
 Github Issue: Fixes https://github.com/mattermost/mattermost-server/issues/16143
JIRA ticket: Fixes https://mattermost.atlassian.net/browse/MM-20431

### CODE REVIEW

File: components/reset_status_modal/index.ts [hunk 1]
```diff
@@ -11,17 +11,17 @@ import {getStatusForUserId} from 'mattermost-redux/selectors/entities/users';
 
 import {autoResetStatus} from 'actions/user_actions.jsx';
 
-import ResetStatusModal from './reset_status_modal.jsx';
+import ResetStatusModal from './reset_status_modal';
 
-function mapStateToProps(state) {
+function mapStateToProps(state: any) {
```
Comment (reflog) [inline components/reset_status_modal/index.ts:16]: Please try to avoid 'any'. See other components that are already typescript for example of properly typing 'state' parameter. (the type is called GlobalState)

File: components/reset_status_modal/index.ts [hunk 2]
```diff
@@ -11,17 +11,17 @@ import {getStatusForUserId} from 'mattermost-redux/selectors/entities/users';
...
     const {currentUserId} = state.entities.users;
     return {
         autoResetPref: get(state, Preferences.CATEGORY_AUTO_RESET_MANUAL_STATUS, currentUserId, ''),
         currentUserStatus: getStatusForUserId(state, currentUserId),
     };
 }
 
-function mapDispatchToProps(dispatch) {
+function mapDispatchToProps(dispatch: any) {
```
Comment (reflog) [inline components/reset_status_modal/index.ts:24]: same

File: components/reset_status_modal/reset_status_modal.tsx [hunk 3]
```diff
@@ -1,58 +1,69 @@
...
+    onHide?: () => void;
+
+    actions: {
 
         /*
-         * The user's preference for whether their status is automatically reset
+         * Function to get and then reset the user's status if needed
          */
-        autoResetPref: PropTypes.string,
+        autoResetStatus: () => any;
```
Comment (reflog) [inline components/reset_status_modal/reset_status_modal.tsx:45]: replace 'any' with proper typing

File: components/reset_status_modal/reset_status_modal.tsx [hunk 4]
```diff
@@ -1,58 +1,69 @@
...
 
         /*
-         * Props value is used to update currentUserStatus
+         * Function to set the status for a user
          */
-        currentUserStatus: PropTypes.string,
+        setStatus: (status: UserStatus) => any;
```
Comment (reflog) [inline components/reset_status_modal/reset_status_modal.tsx:50]: replace 'any' with proper typing

File: components/reset_status_modal/reset_status_modal.tsx [hunk 5]
```diff
@@ -1,58 +1,69 @@
...
 
         /*
-         * Props value is used to reset status from status_dropdown
+         * Function to save user preferences
          */
-        newStatus: PropTypes.string,
+        savePreferences: (userId: string, preferences: Array<PreferenceType>) => any;
```
Comment (reflog) [inline components/reset_status_modal/reset_status_modal.tsx:55]: replace 'any' with proper typing

File: components/reset_status_modal/reset_status_modal.tsx [hunk 6]
```diff
@@ -1,58 +1,69 @@
...
-
-            /*
-             * Function to save user preferences
-             */
-            savePreferences: PropTypes.func.isRequired,
-        }).isRequired,
-    };
+type State = {
+    show: boolean;
+    currentUserStatus: any;
```
Comment (reflog) [inline components/reset_status_modal/reset_status_modal.tsx:61]: replace 'any' with proper typing

File: components/reset_status_modal/reset_status_modal.tsx [hunk 7]
```diff
@@ -1,58 +1,69 @@
...
+    newStatus: string;
+};
 
-    constructor(props) {
+export default class ResetStatusModal extends React.PureComponent<Props, State> {
+    public constructor(props: any) {
```
Comment (reflog) [inline components/reset_status_modal/reset_status_modal.tsx:66]: `props: Props`

File: components/reset_status_modal/reset_status_modal.tsx [hunk 8]
```diff
@@ -62,9 +73,9 @@ export default class ResetStatusModal extends React.PureComponent {
         };
     }
 
-    componentDidMount() {
+    public componentDidMount() {
         this.props.actions.autoResetStatus().then(
-            (status) => {
+            (status: any) => {
```
Comment (reflog) [inline components/reset_status_modal/reset_status_modal.tsx:78]: replace 'any' with proper typing

File: i18n/en.json [hunk 9]
```diff
@@ -3195,7 +3195,6 @@
   "mobile.set_status.offline.icon": "Offline Icon",
   "mobile.set_status.online.icon": "Online Icon",
   "modal.manual_status.ask": "Do not ask me again",
-  "modal.manual_status.auto_responder.message_": "Would you like to switch your status to \"{status}\" and disable Automatic Replies?",
```
Comment (reflog) [inline i18n/en.json:3198]: not sure why this change happened?

**Review by reflog — CHANGES_REQUESTED** [review]
Hi! Thanks for the contribution. Some fixes are still needed. Plus I think en.json and package-lock should not be in this PR

File: i18n/en.json (hunk 9 above)
Comment (kcc343) [reply i18n/en.json:3198]: Thanks for the feedback! When it was testing for i18n check, it was giving me an error on this line. I also did the i18n-extract localization test and it was deleting this line for some reason. I wasn't sure what else to change in my code to prevent that from deleting.

Comment (kcc343) [reply i18n/en.json]: @reflog So I changed all the 'any' types to see if the i18n check would stop deleting this line. Now I'm stumped on what to do here. Any advice?

Comment (catalintomai) [reply i18n/en.json]: @kcc343 , what error were you getting when doing the i18n check? The string is currently used in webapp - can you include here the output of the "make i18n-extract" check?

Comment (kcc343) [reply i18n/en.json]: @catalintomai Here's the output I'm getting from the make i18n-extract


Is it not supposed to be deleting this line or it does need to be deleted?

Comment (catalintomai) [reply i18n/en.json]: I don't know why "make i18n-extract" decides to remove that line. The safe solution is to keep that line and ignore the "make i18n-extract" resulting updates.

**Comment by devinbinnie** [comment]
@kcc343 Do you still plan to finish up this PR?

**Comment by kcc343** [comment]
@devinbinnie Nope I'll let someone else grab this. Thanks for asking!

**Comment by devinbinnie** [comment]
Closing as per above comments.
